service/scaladsl/kafka/server/src/main/scala/com/lightbend/lagom/scaladsl/broker/kafka/LagomKafkaComponents.scala

```diff
@@ -16,6 +16,12 @@ trait LagomKafkaComponents extends LagomKafkaClientComponents {
   def lagomServer: LagomServer
   def offsetStore: OffsetStore
 
+  override def topicPublisherName: Option[String] = super.topicPublisherName match {
+    case Some(other) =>
+      sys.error(s"Cannot provide the kafka topic factory as the default topic publisher since a default topic publisher has already been mixed into this cake: $other")
```

Comment:
What are the circumstances in which this problem would occur?

I can think of two:

1. Hypothetically, someone creates a new message broker implementation and has already included that (unlikely)
2. Someone has a bit of a messy component hierarchy and has already included `LagomKafkaComponents` in an ancestor (more likely)

Maybe the cases should be changed to `case Some(other) if other != "kafka"` and `case _`?

or it could warn if "kafka" is included twice maybe, but I can just imagine that some people will accidentally create this situation by including the trait at multiple points in the hierarchy and then not understand how to fix it.

A lot of people don't read documentation, or don't retain all of it! Another good practice IMO is to link error messages to doc pages that explain how to resolve them.